modules/data.land/R/InventoryGrowthFusion.R

```diff
@@ -224,37 +265,91 @@ model{
+  
+  ########################################################################
+  ###
+  ###        TIME-VARYING
+  ###
+  ########################################################################
+  
+  if(FALSE){ # always false...just for development
     ## DEVEL TESTING FOR TIME VARYING
-    time_varying <- "TminJuly + PrecipDec + TminJuly*PrecipDec"
+    #time_varying <- "TminJuly + PrecipDec + TminJuly*PrecipDec"
+    time_varying <- "tmax_Jun + ppt_Dec + tmax_Jun*ppt_Dec" 
     time_data <- list(TminJuly = matrix(0,4,4),PrecipDec = matrix(1,4,4))
   }
   
-  ## Time-varying covariates
   if(!is.null(time_varying)){
     if (is.null(time_data)) {
-      print("time_varying formula provided but time_data is absent:", time_varying)
+      PEcAn.utils::logger.error("time_varying formula provided but time_data is absent:", time_varying)
     }
+    Xt.priors <- ""
     
     ## parse equation into variable names
     t_vars <- gsub(" ","",unlist(strsplit(time_varying,"+",fixed=TRUE))) ## split on +, remove whitespace
     ## check for interaction terms
-    it_vars <- grep(pattern = "*",x=t_vars,fixed = TRUE)
-      ## need to deal with interactions with fixed variables
-      ## will get really nasty if interactions are with catagorical variables
-      ## need to create new data matrices on the fly
+    it_vars <- t_vars[grep(pattern = "*",x=t_vars,fixed = TRUE)]
+    if(length(it_vars) > 0){
+      t_vars <- t_vars[!(t_vars %in% it_vars)]
+    } 
+    
+    ## INTERACTIONS WITH TIME-VARYING VARS
+    ## TODO: deal with interactions with catagorical variables
+    ## need to create new data matrices on the fly
+    for(i in seq_along(it_vars)){
+
+      ##is covariate fixed or time varying?
+      covX <- strsplit(it_vars[i],"*",fixed=TRUE)[[1]] 
+      tvar    <- length(grep("[t]",covX[1],fixed=TRUE)) > 0
+      tvar[2] <- length(grep("[t]",covX[2],fixed=TRUE)) > 0
+      myBeta <- "beta"
+      for(j in 1:2){
+        if(j == 2) myBeta <- paste0(myBeta,"_")
+        if(tvar[j]){
+          covX[j] <- sub("[t]","",covX[j],fixed = TRUE)
+          if(!(covX[j] %in% names(data))){
+            ## add cov variables to data object
+            data[[covX[j]]] <- time_varying[[covX[j]]]
+          }
+          myBeta <- paste0(myBeta,covX[j])
+          covX[j] <- paste0(covX[j],"[i,t]")
+        } else {
+          ## variable is fixed
+          if(!(covX[j] %in% names(data))){
+            ## add cov variables to data object
+            data[[covX[j]]] <- cov.data[,covX[j]]
+          }
+          myBeta <- paste0(myBeta,covX[j])
+          covX[j] <- paste0(covX[j],"[i]")
+        } ## end fixed or time varying
+        
+      } ## end building beta
+      
+      ## append to process model formula
+      Pformula <- paste(Pformula,
+                        paste0(" + ",myBeta,"*",covX[1],"*",covX[2]))
+      
+      ## priors
+      Xt.priors <- paste0(Xt.priors,
+                          "    ",myBeta,"~dnorm(0,0.001)\n")
```

Comment:
Probably fine for now, but eventually, might be good to let users pass custom priors for these terms?